src/backends/oracle/standard-into-type.cpp

```diff
@@ -232,44 +232,56 @@ void oracle_standard_into_type_backend::pre_fetch()
     }
 }
 
+/* ----------------------------------------------------------------- 
+ * callback function to write buffer to the string.     
+ * works in bytes, not in characters                 
+ * ----------------------------------------------------------------- */
+sb4 callback_to_read_in_bytes
+(
+    dvoid                             *pResultString,  // pointer to std::string to accumulate result
+    CONST dvoid                       *bufxp,          // pointer to buffer from read
+    ub4                                lenp,           // length data in the buffer
+    ub1                                /*piecep*/      // OCI_FIRST_PIECE, OCI_NEXT_PIECE or OCI_LAST_PIECE
+)
+{
+    static_cast<std::string*> ( pResultString )->append ( static_cast<const char *> ( bufxp ), lenp );
+    return OCI_CONTINUE;
+}
+
+
 void oracle::read_from_lob(oracle_session_backend& session,
     OCILobLocator * lobp, std::string & value)
 {
     ub4 len;
 
-    sword res = OCILobGetLength(session.svchp_, session.errhp_,
-        lobp, &len);
+    // result is not a bytes count!!! For character LOBs (CLOBs, NCLOBs) it is the number of characters
+    sword res = OCILobGetLength ( session.svchp_, session.errhp_, lobp, &len );
     if (res != OCI_SUCCESS)
     {
         throw_oracle_soci_error(res, session.errhp_);
     }
 
-    std::vector<char> buf(len);
-
-    if (len != 0)
+    ub4 amtp = 0;
+    ub4 offset = 1;
+    char buf[10240] = {};
```

Comment:
Why do we use both a buffer and a callback? Shouldn't we not need the buffer any longer if we use the callback?

Reply:
oci read into the buffer and callback after